Services/Data/Providers/DatabaseOrdersProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using WPF_Restaurant.Models;
+
+namespace WPF_Restaurant.Services.Data.Providers
+{
+    public class DatabaseOrdersProvider
+    {
+        private readonly RestaurantDbContextFactory _dbContextFactory;
+
+        public DatabaseOrdersProvider(RestaurantDbContextFactory dbContextFactory)
+        {
+            _dbContextFactory = dbContextFactory;
+        }
+
+        public async Task<IEnumerable<Order>> GetAllOrders()
+        {
+            using (var dbContext = _dbContextFactory.CreateDbContext())
+            {
+                var orders = new List<Order>();
+                var ordersDTO = dbContext.Orders.Include(o => o.OrderItems).ThenInclude(x => x.Dish).ToList();
+
+                foreach (var orderFromDb in ordersDTO)
+                {
+                    var dishes = orderFromDb.OrderItems.Select(oi => new Dish(oi.DishId, oi.Dish.Name, oi.Dish.ImagePath, oi.Dish.Recipe, oi.Dish.Ingredients)
```

Comment:
This makes new request to DB and row
`Quantity = orderFromDb.OrderItems.Count(x => x.DishId == oi.DishId)`
makes new request to DB. So you will make 2 subrequests for every order. It will be very slow.
But you already have all data in `ordersDTO`.
I can give you correct code, but it will be better if find it youself.
Hints: 
For each order in `ordersDTO` you can do OrderItems.GroupBy(...) and Count(....) for group.

Reply:
And I think, it's bad choose to calculate Quantity here. Because the showing of Quantity is task of ViewModel.
And when you will show order in rigth side of cooker view you will need to build list of Dishes without Quantity. 
![image](https://user-images.githubusercontent.com/68640750/161521344-769015db-f8bf-45b0-a823-acc212d36184.png)
I mean you collapse the order items to entity with Quantity and then you will need to uncollapse they.
My advice - return Order with OrderItems as is without Quantity, and only into OrderViewModel collapse them to OrderItemViewModel with Quantity.